### ISSUE TITLE
Code Table Request - DGR: Division of Genomic Resources (MSB)

### ISSUE THREAD

**Issue by Jegelewicz** [ArctosDB/arctos#5773]
## Instructions

This is a template to facilitate communication with the Arctos Code Table Committee. Submit a separate request for each relevant value. This form is appropriate for exploring how data may best be stored, for adding vocabulary, or for updating existing definitions.

Reviewing documentation before proceeding will result in a more enjoyable experience.

* [Issue Documentation](http://handbook.arctosdb.org/how_to/How-to-Use-Issues-in-Arctos.html)
* [Code Table Documentation](https://handbook.arctosdb.org/how_to/How-To-Manage-Code-Table-Requests.html)
* [Video Tutorial - Submit a Code Table Request](https://youtu.be/t2jHbsRA3lk)

------------------------------

## Initial Request

**Goal**: Describe what you're trying to accomplish. This is the only necessary step to start this process. The Committee is available to assist with all other steps. Please clearly indicate any uncertainty or desired guidance if you proceed beyond this step.

Replace ambiguous other ID with generic other ID type and issued by agent

> All  [DGR: Division of Genomic Resources (MSB) ](https://arctos.database.museum/info/ctDocumentation.cfm?table=ctcoll_other_id_type#dgr__division_of_genomic_resources__msb_) should be replaced with other ID type = other identifier and issued by agent [Museum of Southwestern Biology, Division of Genomic Resources](https://arctos.database.museum/agents.cfm?agent_id=21298561)

**Proposed Value**: Proposed new value. This should be clear and compatible with similar values in the relevant table and across Arctos. 

N/A

**Proposed Definition**: Clear, complete, non-collection-type-specific **functional** definition of the value. Avoid discipline-specific terminology if possible, include parenthetically if unavoidable.

N/A

**Context**: Describe why this new value is necessary and existing values are not.
See #4604 

**Table**: Code Tables are http://arctos.database.museum/info/ctDocumentation.cfm. Link to the specific table or value. This may involve multiple tables and will control datatype for Attributes. OtherID requests require BaseURL (and example) or explanation. Please ask for assistance if unsure.

https://arctos.database.museum/info/ctDocumentation.cfm?table=ctcoll_other_id_type

**Collection type**: Some code tables contain collection-type-specific values. ``collection_cde`` may be found from https://arctos.database.museum/home.cfm

N/A

**Priority**: Please describe the urgency and/or choose a priority-label to the right. You should expect a response within two working days, and may utilize [Arctos Contacts](https://arctosdb.org/contacts/) if you feel response is lacking.

Normal

**Available for Public View**: Most data are by default publicly available. Describe any necessary access restrictions. 

**Project**: Add the issue to the [Code Table Management Project](https://github.com/ArctosDB/arctos/projects/13#card-31628184).

**Discussion**: Please reach out to anyone who might be affected by this change. Leave a comment or add this to the Committee agenda if you believe more focused conversation is necessary.

## Approval

All of the following must be checked before this may proceed.

The [How-To Document](https://handbook.arctosdb.org/how_to/How-To-Manage-Code-Table-Requests.html) should be followed. Pay particular attention to terminology (with emphasis on consistency) and documentation (with emphasis on functionality).

- [x] Code Table Administrator[1] - check and initial, comment, or thumbs-up to indicate that the request complies with the how-to documentation and has your approval
- [ ] Code Table Administrator[2] - check and initial, comment, or thumbs-up to indicate that the request complies with the how-to documentation and has your approval
- [ ] DBA - The request is functionally acceptable. The term is not a functional duplicate, and is compatible with existing data and code.
- [ ] DBA - Appropriate code or handlers are in place as necessary. (ID_References, Media Relationships, Encumbrances, etc. require particular attention)

## Rejection

If you believe this request should not proceed, explain why here. Suggest any changes that would make the change acceptable, alternate (usually existing) paths to the same goals, etc. 

1. Can a suitable solution be found here? If not, proceed to (2)
2. Can a suitable solution be found by Code Table Committee discussion? If not, proceed to (3)
3. Take the discussion to a monthly Arctos Working Group meeting for final resolution.

## Implementation

Once all of the Approval Checklist is appropriately checked and there are no Rejection comments, or in special circumstances by decree of the Arctos Working Group, the change may be made.

- [ ] Review everything one last time.  Ensure the How-To has been followed. Ensure all checks have been made by appropriate personnel. 

- [ ] Make changes as described above. Ensure the URL of this Issue is included in the definition. 

Close this Issue.

**DO NOT** modify Arctos Authorities in any way before all points in this Issue have been fully addressed; data loss may result.

## Special Exemptions

In very specific cases and by prior approval of The Committee, the approval process may be skipped, and implementation requirements may be slightly altered. Please note here if you are proceeding under one of these use cases.

1. Adding an existing term to additional collection types may proceed immediately and without discussion, but doing so may also subject users to future cleanup efforts. If time allows, please review the term and definition as part of this step.
2. The Committee may grant special access on particular tables to particular users. This should be exercised with great caution only after several smooth test cases, and generally limited to "taxonomy-like" data such as International Commission on Stratigraphy terminology.

**Comment by dustymc** [comment]
I will plan on proceeding with this about 2023-03-23 if there are no objections.

I will proceed immediately upon approval of each of the involved collections.


Data:

[temp_dgr_division_of_genomic_resources_msb_.csv.zip](https://github.com/ArctosDB/arctos/files/10934519/temp_dgr_division_of_genomic_resources_msb_.csv.zip)

Summary:

```

 guid_prefix | numrecs 
-------------+---------
 MSB:Fish    |      10
 MSB:Mamm    |   37978
 MSB:Para    |       1


```


Users:


 @campmlc
 @AdrienneRaniszewski
 @jldunnum
 @msbparasites


See also https://github.com/ArctosDB/arctos/issues/5771

**Comment by Jegelewicz** [comment]
@campmlc can we proceed with this one?

**Comment by campmlc** [comment]
No, these are valid Arctos guids, most of these are used in same individual as relationships, they need to be carefully evaluated first. Too much legacy baggage that needs to be resolved first with an overall master plan and some dedicated attention.

**Comment by Jegelewicz** [comment]
> these are valid Arctos guids

Then can we change them to DGR:Mamm ?

**Comment by campmlc** [comment]
Let me check the fish and para records when I'm back at my computer.

**Comment by dustymc** [comment]
>these are valid Arctos guids

This cannot be true, this identifier is "something about DGR, probably" and would not (or should not) have been used if better resolution were possible.

**Comment by Jegelewicz** [comment]
OK - I made a list of all the DGR GUID that these should be converted to, but I cannot check to see if they exist in Arctos (the list is too long). @dustymc can you check them?

[DGR cleanup.csv](https://github.com/ArctosDB/arctos/files/11559911/DGR.cleanup.csv)

**Comment by campmlc** [comment]
Some of these will have been deleted from the DGR catalog, and those will need to remain as institutional identifiers assigned by MSB Division of Genomic Resources.
All this legacy conversion was already in progress before I started at MSB, and so there is no reason at all to assume that these identifiers "would not (or should not) have been used if better resolution were possible."

**Comment by dustymc** [comment]
>Some
```

arctosprod@arctos>> create table temp_dgr_99876 as select * from  temp_cache.temp_dlm_uptbl;
SELECT 37989
Time: 46.661 ms
arctosprod@arctos>> alter table temp_dgr_99876 add flatguid varchar;
ALTER TABLE
Time: 0.830 ms
arctosprod@arctos>> update temp_dgr_99876 set flatguid=flat.guid from flat where temp_dgr_99876.guid=flat.guid;
UPDATE 72

```
[temp_dgr_99876.csv.zip](https://github.com/ArctosDB/arctos/files/11559953/temp_dgr_99876.csv.zip)


Still don't understand why we must complete impossible tasks that have been ignored for decades before moving to a more-capable system...

**Comment by campmlc** [comment]
For these data all can be converted to institutional catalog number issued by MSB DGR with relationship self.

We would all feel much better about this process if we didn't think this change had even the slightest possibility of making fixing these impossible tasks that have been ignored for decades even more difficult to resolve. None of have the ability to spend the hours needed to determine what will and what will not be adversely affected, repeated reassurences to the contrary notwithstanding.
The georef bot experience has made everyone nervous about massive changes to data without opportunity for review. If we all had sufficient staffing and funding, we could deal with legacy problems and get them fixed in real time. Instead, we are just trying to avoid making bulk changes without fully understanding them. And from the collections perspective, this is being pushed through quickly and piecemeal, issue by issue, with no coherent view of the outcome and no ability to review in a test box first.
@mkoo

**Comment by mkoo** [comment]
obsolete with changes to agents and will close
